app/models/restaurant.rb

```diff
@@ -0,0 +1,17 @@
+class Restaurant < ApplicationRecord
+
+  after_initialize :set_default_values
+
+  validates :name, presence: true, uniqueness: true
+  #TODO: should floats be allowed in rating? is it a stars format or a grade format?
+  validates :rating, numericality: { only_integer: true }, :inclusion => 0..3
+  validates :delivery_time, :numericality => { :only_integer => true, :greater_than_or_equal_to => 0 }
+
```

Comment:
You can also validate the "tenbis" value:

`validates :tenbis, inclusion: [true, false]`